test/saithrift/tests/copp_tests.py

```diff
@@ -0,0 +1,604 @@
+        sai_thrift_create_hostif_trap(
+            client=self.client, 
+            trap_id=SAI_HOSTIF_TRAP_ID_DHCP, 
+            action=SAI_PACKET_ACTION_TRAP, 
+            channel=SAI_HOSTIF_TRAP_CHANNEL_NETDEV, 
+            trap_group_id=trap_group)
+        return
+
+    def runTest(self):
+        self.run_suite()
+
+    def contruct_packet(self, port_number):
+        src_mac = self.mac_map[port_number]
+        packet = simple_udp_packet(pktlen=100,
+                          eth_dst='ff:ff:ff:ff:ff:ff',
+                          eth_src=src_mac,
+                          dl_vlan_enable=False,
+                          vlan_vid=0,
+                          vlan_pcp=0,
+                          dl_vlan_cfi=0,
+                          ip_src='0.0.0.0',
+                          ip_dst='255.255.255.255',
+                          ip_tos=0,
+                          ip_ttl=64,
+                          udp_sport=68,
+                          udp_dport=67,
+                          ip_ihl=None,
+                          ip_options=False,
+                          with_udp_chksum=True
+                          )
+
+        return packet
+
+
+class LLDPTest(NoPolicyTest):
+    def __init__(self):
+        NoPolicyTest.__init__(self)
+        return
+
+    def setUp(self):
+        ControlPlaneBaseTest.setUp(self)
+        trap_group = sai_thrift_create_hostif_trap_group(self.client, queue_id=4)
+        self.trap_groups.append(trap_group)
+        sai_thrift_create_hostif_trap(
+            client=self.client, 
+            trap_id=SAI_HOSTIF_TRAP_ID_LLDP, 
+            action=SAI_PACKET_ACTION_TRAP, 
+            channel=SAI_HOSTIF_TRAP_CHANNEL_NETDEV, 
+            trap_group_id=trap_group)
+
+        return
+
+    def runTest(self):
+        self.run_suite()
+        return
+
+    def contruct_packet(self, port_number):
+
+        src_mac = self.mac_map[port_number]
+        packet = simple_eth_packet(
+                       eth_dst='01:80:c2:00:00:0e',
+                       eth_src=src_mac,
+                       eth_type=0x88cc
+                 )
+        return packet
+
+class BGPTest(NoPolicyTest):
+    def __init__(self):
+        NoPolicyTest.__init__(self)
+
+    def setUp(self):
+        ControlPlaneBaseTest.setUp(self)
+        
+        trap_group = sai_thrift_create_hostif_trap_group(self.client, queue_id=4)
+        self.trap_groups.append(trap_group)
+        sai_thrift_create_hostif_trap(
+            client=self.client, 
+            trap_id=SAI_HOSTIF_TRAP_ID_BGP, 
+            action=SAI_PACKET_ACTION_TRAP, 
+            channel=SAI_HOSTIF_TRAP_CHANNEL_NETDEV, 
+            trap_group_id=trap_group)
+
+    def runTest(self):
+        self.run_suite()
+
+    def contruct_packet(self, port_number):
+        dst_mac = self.test_params['dst_mac']
+        dst_ip = self.peerip[port_number]
+        packet = simple_tcp_packet(
+                      eth_dst=dst_mac,
```

Comment:
Can't use router_mac, on MLNX it is not supported to change SAI_SWITCH_ATTR_SRC_MAC_ADDRESS, which is used to push the value of router_mac in swith.py::switch_init().